lib/routes-utils.js

```diff
@@ -0,0 +1,12 @@
+export const mapRouteToURL = routeObj => {
+  if (routeObj.pathname === undefined) {
+    return null;
+  } else if (routeObj.pathname === '/expenses') {
+    const { collectiveSlug, ...queryURL } = routeObj.query;
+    const queryURLstr = new URLSearchParams(queryURL).toString();
+
+    return `${collectiveSlug}/expenses?${queryURLstr}`;
+  } else {
```

Comment:
What about the other paths (`transactions`, `expenses`, dashboard, etc.)? Is there no generic solution that would not require adding custom rules for each path?

Reply:
> What about the other paths (`transactions`, `expenses`, dashboard, etc.)?

Alright, I've just tested `transactions` and noticed it has the same problem that `expenses` used to have. I'll start working on fixing it later today. `dashboard` seems to be a bit trickier to test, but I'll look into that too. 

> Is there no generic solution that would not require adding custom rules for each path?

I don't think there's any way to go around this. The old `Router` from `../server/pages` does exactly this too. The rules there are specified in a special syntax which does make each rule a bit more compact, but the principle is the same and they are all specified manually.